File: pdu/javascript/sample_action_msgs/pdu_cdr_conv_FibonacciActionResponse.js
import { FibonacciActionResponse } from './pdu_jstype_FibonacciActionResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { ActionResponseHeader } from '../hako_action_msgs/pdu_jstype_ActionResponseHeader.js';
import { PduActionResponseHeaderConverter } from '../hako_action_msgs/pdu_cdr_conv_ActionResponseHeader.js';
import { FibonacciResult } from '../sample_action_msgs/pdu_jstype_FibonacciResult.js';
import { PduFibonacciResultConverter } from '../sample_action_msgs/pdu_cdr_conv_FibonacciResult.js';


export class PduFibonacciActionResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { FibonacciActionResponse } src
     */
    static to_cdr_body(writer, src) {
        PduActionResponseHeaderConverter.to_cdr_body(writer, src.header);
        PduFibonacciResultConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { FibonacciActionResponse } dst
     * @returns { FibonacciActionResponse }
     */
    static cdr_body_to_js(reader, dst) {
        PduActionResponseHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduFibonacciResultConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { FibonacciActionResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { FibonacciActionResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new FibonacciActionResponse());
    }
}

File: pdu/python/sample_action_msgs/pdu_cdr_conv_FibonacciActionResponse.py
from .pdu_pytype_FibonacciActionResponse import FibonacciActionResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_action_msgs.pdu_cdr_conv_ActionResponseHeader import *
from ..sample_action_msgs.pdu_cdr_conv_FibonacciResult import *



def py_to_cdr_body_FibonacciActionResponse(writer: CdrWriter, src: FibonacciActionResponse):
    py_to_cdr_body_ActionResponseHeader(writer, src.header)
    py_to_cdr_body_FibonacciResult(writer, src.body)


def cdr_body_to_py_FibonacciActionResponse(reader: CdrReader, dst: FibonacciActionResponse):
    cdr_body_to_py_ActionResponseHeader(reader, dst.header)
    cdr_body_to_py_FibonacciResult(reader, dst.body)
    return dst


def py_to_cdr_FibonacciActionResponse(src: FibonacciActionResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_FibonacciActionResponse(writer, src)
    return writer.bytes()


def cdr_to_py_FibonacciActionResponse(cdr_payload) -> FibonacciActionResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = FibonacciActionResponse()
    return cdr_body_to_py_FibonacciActionResponse(reader, dst)
